### DISCUSSION TITLE
[Feature] iCloud Shared Library Control Parity

### DISCUSSION THREAD

**Discussion by undaunt** [immich-app/immich#6478] [Feature Request]
### The feature

I'd like for my wife and I to both be able to delete, favorite, update metadata, or otherwise manage items that are uploaded from our iCloud shared photo library under both of our accounts without the assets being duplicated. I'd also like for both of us to be able to use the mobile client to upload.

**Initially**
I bulk uploaded my entire library via CLI, and am using the mobile app, and have all of our shared assets in *my* library. She only has her personal iOS photos in her library, and we both have partner sharing enabled in Immich. (Basically, I uploaded my PrimarySync and SharedSyncXXXXXXXX to me, and her PrimarySync library to her via icloud-photos-downloader.)

**Challenges**
This sort of works and allows us to both see all of our photos, but also precludes her from using the mobile app. It would create duplicates of everything in the iCloud shared library if she synced via mobile since we cannot selectively have her upload everything only in her non-shared library.

Technically, since the shared photos are 'mine' in Immich she also cannot edit or remove them in any way. Also, since only I am using the mobile app, none of her favorites are synced as she frequently manages favorites in iOS photos.

If I were to go another route and have a 3rd user account that we both log into manage the shared library, I don't know how I'd be able to exclude only shared library photos from mobile (as they are not an "album" on my phone), thus precluding either of us from using the mobile app and/or getting favorites synced.

I've seen some other requests about partners and sharing of specific requests, face data, etc. but I didn't see this specific request, so apologies if it is a duplicate.

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by ericswpark** [comment]
I think before Immich can implement filtering iCloud shared libraries on the mobile side it should implement a shared library feature so that assets can properly be merged between different user accounts.

**Comment by DanielLaberge** [comment]
I am in a similar situation, researching how to best upload our iCloud libraries, including a shared library, without creating duplicates in Immich.

The simplest solution I can imagine might be: 
- A server setting to detect duplicates ACROSS two libraries; It would allow marking which library is the "main" one.
- When duplicates are found in the second library, delete the copy, but keep a reference to the asset in the main library, allowing the second user to see, organize or delete the asset from their library.
- A label in the UI could indicate a "Shared Asset" status.

Does this sound reasonable to everyone?

**Comment by yozik04** [comment]
Suggestions to implement Shared Library feature effectively:

- When creating a shared library, all content within it is co-owned by both members. Both you and your partner have equal rights to manage the library without any restrictions.
- iOS offers a highly useful Shared Library feature. It would be beneficial if Immich could identify the metadata indicating whether a photo is personal or shared and upload it to the appropriate library within Immich. However, it is unclear if Apple provides access to this metadata.
- Photos uploaded to the shared library should be checked for duplicates, because these could be uploaded from multiple devices.
- In the timeline, photos could display an icon indicating their presence in the shared library.
- When moving photos or albums to the shared library, they should be transferred and removed from your personal library.

Does this sound reasonable?

**Comment by PullingBerlin** [comment]
For me it would also be highly helpful if there was a feature to filter between shared and personal library in the upload feature in the iOS app. My wife and me also work with the shared library feature a lot and this would make it way easier to import photos without duplicates.
